docs/src/proposals/transactions-v2.md

```diff
@@ -158,15 +157,51 @@ pub struct AddressMap {
   #[serde(with = "short_vec")]
   pub entries: Vec<u8>,
 }
+
+#[derive(Serialize, Deserialize)]
+pub struct CompiledInstructionV2 {
+  /// unchanged
+  pub program_id_index: u8,
+
+  /// unchanged
+  #[serde(with = "short_vec")]
+  pub accounts: Vec<u8>,
+
+  /// Only programs in this list may be invoked by the outer instruction
+  #[serde(with = "short_vec")]
+  pub inner_programs: Vec<u8>,
+
+  /// Dependent accounts do not need to be mapped to the memory of the outer
+  /// instruction's execution context. They are available for passing to inner
+  /// instructions
+  #[serde(with = "short_vec")]
+  pub inner_program_accounts: Vec<u8>,
```

Comment:
I think we can avoid this additional churn to solve the problem described in this pr's description by providing yet another pathway (#17796 ) to access to tx's accounts in addition to the entrypoint one.

Reply:
This can serve two purposes. One is avoiding the serialization overhead. The other (not mentioned in the PR description) is a stronger framework for isolating the scope of what an instruction can do. If we allow an instruction to dynamically load any account from the transaction, a malicious program might have enough resources for more damage. Imagine for instance, a program dynamically loads the fee payer signer account.